### PR TITLE
Update for Prometheus Community

### PR DESCRIPTION
**Pull request by SuperQ**
Add standard Prometheus build setup
* CircleCI config
* Makefile
* Go modules
* Golang-CI Lint
* promu config
* Remove /vendor
* Remove mage build
* Update READMEs

https://github.com/prometheus-community/postgres_exporter/issues/478

Fixes: https://github.com/prometheus-community/postgres_exporter/issues/470

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
After this is done, we can work on re-adding integration tests. We do similar things in the [mysqld_exporter CircleCI config](https://github.com/prometheus/mysqld_exporter/blob/9a43c02fa1a593eb9e1d5308a796b282e17e9ccd/.circleci/config.yml#L71-L81).

**Comment by SuperQ** [comment]
Ping @wrouesnel

**Review by roidelapluie — APPROVED** [review]

File: Makefile.common [hunk 1]
```diff
@@ -69,12 +69,21 @@ else
 	GO_BUILD_PLATFORM ?= $(GOHOSTOS)-$(GOHOSTARCH)
 endif
 
-PROMU_VERSION ?= 0.4.0
+GOTEST := $(GO) test
```
Comment (roidelapluie) [inline Makefile.common:72]: lol at git seeing it as a rename :)

**Comment by roidelapluie** [comment]
Can you please give a second pass at the README ? still mentions vendoring and mage.

**Comment by SuperQ** [comment]
Ok, I updated the README some more.

**Review by roidelapluie — APPROVED** [review]
I don't even think that we need to mention vendoring at all but LGTM.

**Pull request merged**